### DISCUSSION TITLE
How to reinitialize sensor

### DISCUSSION THREAD

**Discussion by Vince-LD** [n-elia/MAX30102-MicroPython-driver#14] [Help]
Hello,
Thank you for your work on this great library. It runs really well for me!
However, I wanted to know how to reset the sensor without having to restart the whole microcontroller.

When I launch my main.py for the first time, everything runs fine. Yet, if I start it a second time, I get this error:
```
Traceback (most recent call last):
  File "<stdin>", line 47, in <module>
  File "max30102.py", line 194, in setup_sensor
  File "max30102.py", line 286, in soft_reset
  File "max30102.py", line 595, in set_bitmask
  File "max30102.py", line 586, in i2c_read_register
OSError: [Errno 19] ENODEV
```
With the previous version of the library, the error was actually `Cannot find sensor on the I2C bus. `. I'm sure I am doing something wrong but i don't know what. Do you have an idea on how to avoid this error in the sensor setup when restarting a script?
I'm running on an ESP32 Devkit 1.

**Comment by n-elia** [comment]
Hi, thank you for sharing your thoughts and experience!

Your issue may be related to the "I2C read issue" point discussed in [Tested platforms and known issues](https://github.com/n-elia/MAX30102-MicroPython-driver#tested-platforms-and-known-issues). There you can find some references to similar issues discussions.

Are you experiencing the issue after a soft reset or a hard reset? Did you write your main.py such that you're sure that a `i2c.scan()` is always performed after a reset, as in [line 25](https://github.com/n-elia/MAX30102-MicroPython-driver/blob/main/example/main.py#L25) of the example script?

**Reply by Vince-LD** [reply]
I added the code snippet you suggested and nothing changes. the sensor is detcted and connected.
The idea is that I don't reset the ESP, I just re-run my main.py. The device is still on the I2C bus, but when `setup_sensor() `is called, it throws the error I copied in my first message...
If I restart the esp by pressing the RST button, I don't have the issue. But in my application (arm band), the ESP will basically never be turned off.

**Reply by n-elia** [reply]
Ok, than let's first point out that the error you get is thrown by the SoftI2C instance that (I guess) you're using to feed the MAX30102 library. I removed the "Cannot find sensor on the I2C bus." message because I want the library to be independant from the I2C library chosen by the user.

So, the problem depends on how you're using the SoftI2C instance. May you share your code, or at least explain what do you mean by "re-run my main.py" ? By default, main.py is run after a hard reset. Is your I2C instance preserved between those "runs"?
